src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -103,14 +125,90 @@ def skip_table(self, backend: StatementExecutionBackend, schema: str, table: str
         except BadRequest as br:
             logger.error(br)
 
-    def skip_schema(self, backend: StatementExecutionBackend, schema: str):
+    def skip_schema(self, schema: str):
         # Marks a schema to be skipped in the migration process by applying a table property
         try:
-            backend.execute(f"ALTER SCHEMA `{schema}` SET DBPROPERTIES('{self.UCX_SKIP_PROPERTY}' = true)")
+            self._backend.execute(f"ALTER SCHEMA `{schema}` SET DBPROPERTIES('{self.UCX_SKIP_PROPERTY}' = true)")
         except NotFound as nf:
             if "[SCHEMA_NOT_FOUND]" in str(nf):
                 logger.error(f"Failed to apply skip marker for Schema {schema}. Schema not found.")
             else:
                 logger.error(nf)
         except BadRequest as br:
             logger.error(br)
+
+    def _get_upgraded_tables(self) -> dict[str, TableInfo]:
+        upgraded_tables = {}
+        for catalog in self._ws.catalogs.list():
+            if not catalog.name:
+                continue
+            for schema in self._ws.schemas.list(catalog_name=catalog.name):
+                if not schema.name:
+                    continue
+                for table in self._ws.tables.list(catalog_name=catalog.name, schema_name=schema.name):
+                    if table.properties is not None and "upgraded_from" in table.properties:
+                        upgraded_tables[table.properties["upgraded_from"].lower()] = table
+        return upgraded_tables
+
+    def _get_databases_in_scope(self, databases: set[str]):
+        tasks = []
+        for database in databases:
+            tasks.append(partial(self._get_database_in_scope_task, database))
+        return Threads.strict("checking databases for skip property", tasks)
+
+    def _get_database_in_scope_task(self, database: str) -> str | None:
+        describe = {}
+        for value in self._backend.fetch(f"DESCRIBE SCHEMA EXTENDED {database}"):
+            describe[value["database_description_item"]] = value["database_description_value"]
+        if self.UCX_SKIP_PROPERTY in TablesCrawler.parse_database_props(describe.get("Properties", "").lower()):
+            logger.info(f"Database {database} is marked to be skipped")
+            return None
+        return database
+
+    def _get_tables_in_scope(self, rules: list[Rule], databases_in_scope: set[str], crawled_tables: list[Table]):
+        crawled_tables_keys = {crawled_table.key: crawled_table for crawled_table in crawled_tables}
```

Comment:
```suggestion
    def _get_tables_in_scope(self, tables_crawler: TablesCrawler):
        rules = self.load()
        databases_in_scope = self._get_databases_in_scope({_.src_schema for _ in rules})
        crawled_tables_keys = {_.key: _ for _ in tables_crawler.snapshot()}
```


p.s. `crawled_tables_keys` does not seem to be correct noun. suggest to rename it by `table_by_key` or something

Reply:
KK